controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1025,13 +1030,24 @@ func TestDeleteNSTemplateSet(t *testing.T) {
 		err = r.Client.Delete(context.TODO(), ns)
 		require.NoError(t, err)
 
-		// deletion of ns would trigger another reconcile
+		// deletion of firstNS would trigger another reconcile deleting secondNS
 		result, err = r.Reconcile(context.TODO(), req)
 		require.Empty(t, result)
 		require.NoError(t, err)
 
 		// get the first namespace and check it IS deleted
 		AssertThatNamespace(t, firstNSName, r.Client).DoesNotExist()
+
+		// Check that nsTemplateSet still has finalizer
+		AssertThatNSTemplateSet(t, namespaceName, username, r.Client).
+			HasFinalizer().HasConditions(Terminating())
+
+		// deletion of secondNS would trigger another reconcile
+		result, err = r.Reconcile(context.TODO(), req)
+		require.Empty(t, result)
+		require.NoError(t, err)
```

Comment:
Is this one reconcile is enough here? Don't we have to call another reconcile?
The first one will delete the namespace (`client.Delete()` will be called) and the second one will make sure there is no namespaces left so it can delete the finalizer?

Reply:
Ideally it should, but since delete is triggered in the test itself instead of reconcile, the entire process takes one reconcile less.